dev/tools/gen_keycodes/lib/logical_key_data.dart

```diff
@@ -0,0 +1,596 @@
+        if (keyLabel != null && !entry.constantName.startsWith('numpad')) {
+          return kUnicodePlane | (keyLabel.codeUnitAt(0) & kValueMask);
+        } else {
+          final PhysicalKeyEntry? physicalEntry = physicalData.tryEntryByName(entry.name);
+          if (physicalEntry != null) {
+            return kHidPlane | (physicalEntry.usbHidCode & kValueMask);
+          }
+        }
+      })();
+      if (value != null)
+        entry.fuchsiaValues.add(value);
+    }
+  }
+
+  // Map Web key to the pair of key names
+  static late final Map<String, _ModifierPair> _chromeModifiers = () {
+    final String rawJson = File(path.join(dataRoot, 'chromium_modifiers.json',)).readAsStringSync();
+    return (json.decode(rawJson) as Map<String, dynamic>).map((String key, dynamic value) {
+      final List<dynamic> pair = value as List<dynamic>;
+      return MapEntry<String, _ModifierPair>(key, _ModifierPair(pair[0] as String, pair[1] as String));
+    });
+  }();
+
+  /// Returns the static map of printable representations.
+  static late final Map<String, String> printable = ((){
+    final String printableKeys = File(path.join(dataRoot, 'printable.json',)).readAsStringSync();
+    return (json.decode(printableKeys) as Map<String, dynamic>)
+      .cast<String, String>();
+  })();
+
+  // Map printable to corresponding numpad key name
+  static late final Map<String, String> _printableToNumpads = () {
+    final String rawJson = File(path.join(dataRoot, 'printable_to_numpads.json',)).readAsStringSync();
+    return (json.decode(rawJson) as Map<String, dynamic>).map((String key, dynamic value) {
+      return MapEntry<String, String>(key, value as String);
+    });
+  }();
+
+  /// Returns the static map of synonym representations.
+  ///
+  /// These include synonyms for keys which don't have printable
+  /// representations, and appear in more than one place on the keyboard (e.g.
+  /// SHIFT, ALT, etc.).
+  static late final Map<String, List<String>> synonyms = ((){
+    final String synonymKeys = File(path.join(dataRoot, 'synonyms.json',)).readAsStringSync();
+    final Map<String, dynamic> dynamicSynonym = json.decode(synonymKeys) as Map<String, dynamic>;
+    return dynamicSynonym.map((String name, dynamic values) {
+      // The keygen and algorithm of macOS relies on synonyms being pairs.
+      // See siblingKeyMap in macos_code_gen.dart.
+      final List<String> names = (values as List<dynamic>).whereType<String>().toList();
+      assert(names.length == 2);
+      return MapEntry<String, List<String>>(name, names);
+    });
+  })();
+}
+
+
+/// A single entry in the key data structure.
+///
+/// Can be read from JSON with the [LogicalKeyEntry.fromJsonMapEntry] constructor, or
+/// written with the [toJson] method.
+class LogicalKeyEntry {
+  /// Creates a single key entry from available data.
+  LogicalKeyEntry({
+    required this.value,
+    required this.name,
+    this.keyLabel,
+  })  : assert(name != null),
```

Comment:
Since this is now null-safe, we don't need these asserts anymore (and since it's not part of user code, it doesn't need to run in mixed mode either). (here, and elsewhere)